app/controllers/admin/payout_reports_controller.rb

```diff
@@ -31,6 +31,31 @@ def notify
     redirect_to admin_payout_reports_path, flash: { notice: "Sending notifications to publishers with disconnected wallets." }
   end
 
+  def upload_settlement_report
+    content = File.read(params[:file].tempfile)
+    json = JSON.parse(content)
+    json.each do |entry|
+      next unless entry["documentId"]
+      invoice = Invoice.find_by(id: entry["documentId"])
+      next unless invoice.present?
+
+      invoice.update(
+        paid: true,
+        payment_date: Date.today,
+        status: "paid",
+        paid_by: current_publisher
+      )
+    end
+
+    Eyeshade::Publishers.new.create_settlement(body: json)
```

Comment:
Would you move this line above `invoice.update` block? Otherwise if Eyeshade throws an error, this would show the invoice as paid when it's not.

Reply:
Ah good catch!
https://github.com/brave-intl/publishers/pull/1660/commits/c26b02ec4f3dfe7db967669299103bd36bce630b